src/search/SearchMessagesScreen.js

```diff
@@ -13,10 +13,16 @@ import { getAuth } from '../account/accountsSelectors';
 
 type Props = {|
   auth: Auth,
+  // Warning: do not add new props without considering their effect on the
+  // behavior of this component's non-React internal state. See comment below.
 |};
 
 type State = {|
+  /** The list of messages returned for the latest query, or `null` if there is
+   *  effectively no "latest query" to have results from.
+   */
   messages: Message[] | null,
+  /** Whether there is currently an active valid network request. */
```

Comment:
This isn't true as of this commit, right? This is the meaning that a later commit gives it.

If you have this commit document it as what it means at this point (or actually even just not document it yet at all -- that'd be fine), and then change the doc comment to this in the same commit that changes the meaning, I think that will help clarify both this and the later commit.

Reply:
> This isn't true as of this commit, right? This is the meaning that a later commit gives it.

That's correct — this was an error in the commit refactoring.